auth_server/src/auth.rs

```diff
@@ -0,0 +1,40 @@
+use actix_web::{
+    dev::ServiceRequest, error::ErrorUnauthorized, http::header::AUTHORIZATION, Error,
+};
+use jsonwebtoken::{decode, DecodingKey, Validation};
+
+pub struct AuthMiddleware;
+
+impl AuthMiddleware {
+    pub fn validate_token(
+        req: ServiceRequest,
+        secret: &str,
+    ) -> Result<ServiceRequest, (Error, ServiceRequest)> {
```

Comment:
Error as tuple is not readable, it is better to create your own error type and pass ServiceRequest inside it.

Reply:
It is the shape of response type which is required by the package I'm using for applying middleware to the application